src/effects/RealtimeEffectManager.cpp

```diff
@@ -225,23 +215,20 @@ void RealtimeEffectManager::RealtimeProcessStart()
             state->GetEffect().RealtimeProcessStart();
       }
    }
-
-   mRealtimeLock.Leave();
 }
 
 //
 // This will be called in a different thread than the main GUI thread.
 //
 size_t RealtimeEffectManager::RealtimeProcess(int group, unsigned chans, float **buffers, size_t numSamples)
 {
-   // Protect ourselves from the main thread
-   mRealtimeLock.Enter();
+   // Protect...
+   std::lock_guard<std::mutex> guard(mLock);
```

Comment:
Shouldn't we avoid such things in the realtime processing thread?

Reply:
That would be good.  Reexamination of why that lock is there, and maybe elimination of it, should be attempted.  But two answers:

It's not the purpose of the pull request to fix this.  It's not worse than before, in which there was also locking.

Also the realtime effect processing will be moved (Issue #2082) from the "real-time" thread to the other worker thread that sends messages across the RingBuffer.  That thread needs to maintain the throughput but has more tolerance for variance in time to process batches.  Locking a mutex or allocating memory are things less urgent to remove from that thread.